ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -0,0 +1,630 @@
+        baseDomain: ${BASE_DOMAIN}
+        metadata:
+          name: ${CLUSTER_NAME}
+        networking:
+          machineCIDR: 139.178.73.0/26
+        platform:
+          vsphere:
+            vCenter: vcsa-ci.vmware.devcluster.openshift.com
+            username: "${VSPHERE_USER}"
+            password: "${VSPHERE_PASSWORD}"
+            datacenter: dc1
+            defaultDatastore: nvme-ds1
+        pullSecret: >
+          ${PULL_SECRET}
+        sshKey: |
+          ${SSH_PUB_KEY}
+        EOF
+        else
+            echo "Unsupported cluster type '${CLUSTER_TYPE}'"
+            exit 1
+        fi
+
+        echo "Creating ignition configs"
+        openshift-install --dir=/tmp/artifacts/installer create ignition-configs &
+        wait "$!"
+
+        cp /tmp/artifacts/installer/bootstrap.ign /srv
+
+        mkdir -p /tmp/tf
+
+        # Copy sample UPI files
+        cp -r /var/lib/openshift-install/upi/${CLUSTER_TYPE}/* /tmp/tf
+
+        # Create terraform.tfvars
+        export BOOTSTRAP_URL="http://${JOB_NAME_SAFE}-bootstrap-exporter-${NAMESPACE}.svc.ci.openshift.org/bootstrap.ign"
+        export MASTER_IGN=$(cat /tmp/artifacts/installer/master.ign)
+        export WORKER_IGN=$(cat /tmp/artifacts/installer/worker.ign)
+
+        cat > /tmp/tf/terraform.tfvars <<-EOF
+          machine_cidr = "139.178.73.0/26"
+
+          vm_template = "rhcos-latest"
+
+          vsphere_cluster = "devel"
+
+          vsphere_datacenter = "dc1"
+
+          vsphere_datastore = "nvme-ds1"
+
+          vsphere_server = "vcsa-ci.vmware.devcluster.openshift.com"
+
+          ipam = "139.178.89.254"
+
+          cluster_id = "${CLUSTER_NAME}"
+
+          base_domain = "${BASE_DOMAIN}"
+
+          cluster_domain = "${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+          bootstrap_ignition_url = "${BOOTSTRAP_URL}"
+
+          // Ignition config for the control plane machines. You should copy the contents of the master.ign generated by the installer.
+          control_plane_ignition = <<END_OF_MASTER_IGNITION
+          ${MASTER_IGN}
+          END_OF_MASTER_IGNITION
+
+          // Ignition config for the compute machines. You should copy the contents of the worker.ign generated by the installer.
+          compute_ignition = <<END_OF_WORKER_IGNITION
+          ${WORKER_IGN}
+          END_OF_WORKER_IGNITION
+        EOF
+
+        # Copy secret tfvars
+        cp ${TFVARS_PATH} /tmp/tf
+
+        cd /tmp/tf
+        echo "Creating infra"
+        terraform init -input=false -no-color &
+        wait "$!"
+
+        terraform apply -auto-approve -no-color &
+        wait "$!"
+
+        echo "Waiting for bootstrap to complete"
+        openshift-install --dir=/tmp/artifacts/installer wait-for bootstrap-complete &
+        wait "$!"
+
+        echo "Bootstrap complete, destroying bootstrap VM"
+        terraform apply -auto-approve -var 'bootstrap_complete=true' -no-color &
+        wait "$!"
+
+        echo "Completing UPI setup"
```

Comment:
Before this step, we should
1) Set the image-registry to use an empty dir for its storage backend.
```
oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"filesystem":{"volumeSource": {"emptyDir":{}}}}}}'
```
2) Approve pending CSRs.
```
oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
```